src/test/java/spoon/generating/CorrectIdentifierTest.java

```diff
@@ -69,4 +69,10 @@ public void correctIdentiferUtfChinese() {
 		CtLocalVariableReference<Object> localVariableRef = new Launcher().getFactory().createLocalVariableReference();
 		assertDoesNotThrow(() -> localVariableRef.setSimpleName("処理"));
 	}
+
+	@Test
+	public void correctSquareBrackets() {
+		CtLocalVariableReference<Object> localVariableRef = new Launcher().getFactory().createLocalVariableReference();
+		assertDoesNotThrow(() -> localVariableRef.setSimpleName("List<String>[]"));
```

Comment:
is that a valid name for a local variable reference? Maybe you mean a type reference instead?

Reply:
It isn't a allowed identifier for a local variable, must be more or less alphanumeric with some exceptions.
We have 1 check for all `setSimpleName` methods. Looking at the decision today i would prefer something like a visitor. Allowing different handling for the cases.

Reply:
OK, @Strum355 could you create a type reference instead?

Reply:
My attempts have been unsuccessful unfortunately, I consistently get NPE as getPackage always returns null and Ive been unable to run the Launcher in classpath mode. Im afraid my knowledge of Spoon only goes sofar :disappointed:

Reply:
Fixed by `CtTypeReference localVariableRef = new Launcher().getFactory().createTypeReference();`

LGTM, OK to merge